pkg/piecestore/rpc/client/readerwriter.go

```diff
@@ -14,18 +15,43 @@ import (
 // StreamWriter creates a StreamWriter for writing data to the piece store server
 type StreamWriter struct {
 	stream pb.PieceStoreRoutes_StoreClient
+	signer *Client // We need this for signing
 }
 
 // Write Piece data to a piece store server upload stream
 func (s *StreamWriter) Write(b []byte) (int, error) {
-	if err := s.stream.Send(&pb.PieceStore{Content: b}); err != nil {
+
+	// TODO: What does the message look like?
+	sig, err := s.signer.sign(context.Background(), []byte{'m', 's', 'g'})
```

Comment:
Does `sign` really needs context? If it does then we should passed something different than `context.Background()`.